[docs] Migrate Expo Router docs form home to Guides (#25646)
expo/expo: docs/pages/router/migrate/from-react-navigation.mdx

Create an **app** directory at the root of your repo, or in a root **src** directory.

> Layout the structure of your app by creating files according to the [creating pages guide](/router/create-pages/). Best practice dictates that you use kebab-case and lowercase letters for filenames.

Reviewer: ```suggestion
Layout the structure of your app by creating files according to the [creating pages guide](/router/create-pages/). Kebab-case and lowercase letters are considered best practice for route filenames.
```
